homeassistant/components/vesync/coordinator.py

```diff
@@ -0,0 +1,42 @@
+"""Representation of VeSync data coordinator."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from pyvesync import VeSync
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import UPDATE_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VeSyncDataCoordinator(DataUpdateCoordinator):
+    """Class representing data coordinator for VeSync devices."""
+
+    def __init__(self, hass: HomeAssistant, manager: VeSync) -> None:
+        """Initialize."""
+        self._manager = manager
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="VeSyncDataCoordinator",
+            update_interval=timedelta(seconds=UPDATE_INTERVAL),
+        )
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Fetch data from API endpoint."""
+
+        try:
+            # Using `update_all_devices` instead of `update` to avoid fetching device list every time.
+            return await self.hass.async_add_executor_job(
+                self._manager.update_all_devices
+            )
+        except Exception as error:
+            raise UpdateFailed(error) from error
```

Comment:
Let's not catch bare exceptions. Please be more specific with what can be raised

Reply:
The call into pyvesync is from device.update > [device.get_details](https://github.com/webdjoe/pyvesync/blob/736c29256a5a8b562ddb1b16ac20e418b3e27f29/src/pyvesync/vesyncfan.py#L387) > Helpers.call_api.

pyvesync catches all exceptions and does not rethrow them. It also performs `get` on dictionaries safely so I don't think we will get an exception. If data was malformed or request failed, then we will just not get any data. I am removing the catch.